src/chatwindow/chatwidget.cpp

```diff
@@ -49,3 +54,41 @@ void ChatWidget::setUsersList(const QStringList &usersList)
     }
     mpUsersList->setSizePolicy(QSizePolicy::Maximum, QSizePolicy::Maximum);
 }
+
+void ChatWidget::setCurrentUserName(const QString &userName)
+{
+    mUserName = userName;
+}
+
+void ChatWidget::displayMessage(const QString &userName,
+                                const QString &message)
+{
+    QString feed = mpFeed->toPlainText();
+    feed += '\n';
+    QString insertion = "<" + userName + "> " + message;
+    feed += insertion;
+    mpFeed->setPlainText(feed);
+}
+
+void ChatWidget::onUserToConnectSelected(QListWidgetItem *item)
+{
+    QString userName = item->text();
+    int position = userName.lastIndexOf(QChar{':'});
+    emit userToConnectSelected(userName.mid(position + 2));
```

Comment:
why (position + 2)?

Reply:
We process here the following string:
"Connect to: userName"
I cannot begin search from the end as user name might consist of several words or any combination of characters